projects/fsastorefrontlib/src/cms-components/my-account/order/order-details/order-detail-items/order-detail-items.component.spec.ts

```diff
@@ -0,0 +1,127 @@
+import { waitForAsync, ComponentFixture, TestBed } from '@angular/core/testing';
+import { I18nTestingModule, Order } from '@spartacus/core';
+import { OrderDetailsService } from '@spartacus/storefront';
+import { of } from 'rxjs';
+import { FSOrderDetailItemsComponent } from './order-detail-items.component';
+
+const mockProduct = { product: { code: 'test' } };
+
+const mockOrder: Order = {
+  code: '1',
+  statusDisplay: 'Shipped',
+  deliveryAddress: {
+    firstName: 'John',
+    lastName: 'Smith',
+    line1: 'Buckingham Street 5',
+    line2: '1A',
+    phone: '(+11) 111 111 111',
+    postalCode: 'MA8902',
+    town: 'London',
+    country: {
+      isocode: 'UK',
+    },
+  },
+  deliveryMode: {
+    name: 'Standard order-detail-shipping',
+    description: '3-5 days',
+  },
+  paymentInfo: {
+    accountHolderName: 'John Smith',
+    cardNumber: '************6206',
+    expiryMonth: '12',
+    expiryYear: '2026',
+    cardType: {
+      name: 'Visa',
+    },
+    billingAddress: {
+      firstName: 'John',
+      lastName: 'Smith',
+      line1: 'Buckingham Street 5',
+      line2: '1A',
+      phone: '(+11) 111 111 111',
+      postalCode: 'MA8902',
+      town: 'London',
+      country: {
+        isocode: 'UK',
+      },
+    },
+  },
+  created: new Date('2019-02-11T13:02:58+0000'),
+  consignments: [
+    {
+      code: 'a00000341',
+      status: 'READY',
+      statusDate: new Date('2019-02-11T13:05:12+0000'),
+      entries: [{ orderEntry: {}, quantity: 1, shippedQuantity: 1 }],
+    },
+    {
+      code: 'a00000343',
+      status: 'DELIVERY_COMPLETED',
+      statusDate: new Date('2019-02-11T13:05:12+0000'),
+      entries: [{ orderEntry: mockProduct, quantity: 4, shippedQuantity: 4 }],
+    },
+    {
+      code: 'a00000348',
+      status: 'PICKUP_COMPLETE',
+      statusDate: new Date('2019-02-11T13:05:12+0000'),
+      entries: [{ orderEntry: {}, quantity: 4, shippedQuantity: 4 }],
+    },
+    {
+      code: 'a00000342',
+      status: 'CANCELLED',
+      statusDate: new Date('2019-02-11T13:05:12+0000'),
+      entries: [{ orderEntry: {}, quantity: 0, shippedQuantity: 0 }],
+    },
+    {
+      code: 'a00000349',
+      status: 'OTHERS',
+      statusDate: new Date('2019-02-11T13:05:12+0000'),
+      entries: [{ orderEntry: {}, quantity: 1, shippedQuantity: 1 }],
```

Comment:
Regarding the consignments, I would personally go with only 2 entry of consignments because it reflects FSA more clearly and shipped quantity should be 1 as we cant add more products like in B2C.